external_data/task_functions.py

```diff
@@ -0,0 +1,534 @@
+    ]
+    colter_elus.to_csv(data_dir + "elu-new.csv", index=False)
+
+
+def preprocess_rge_data(
+    data_dir,
+) -> None:
+    from typing import List
+
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+    arr: List[str] = []
+    r = requests.get(
+        "https://data.ademe.fr/data-fair/api/v1/datasets/liste-des-entreprises-rge-2/"
+        "lines?size=10000&select=siret%2Ccode_qualification"
+    )
+    data = r.json()
+    arr = arr + data["results"]
+    cpt = 0
+    print(data)
+    while "next" in data:
+        cpt = cpt + 1
+        r = requests.get(data["next"])
+        data = r.json()
+        arr = arr + data["results"]
+    df = pd.DataFrame(arr)
+    df["siren"] = df["siret"].str[:9]
+    res = (
+        df.groupby(["siren"])["code_qualification"]
+        .apply(list)
+        .reset_index(name="liste_rge")
+    )
+    res = res[["siren", "liste_rge"]]
+
+    res.to_csv(data_dir + "rge-new.csv", index=False)
+
+
+def preprocess_spectacle_data(
+    data_dir,
+) -> None:
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+
+    r = requests.get(
+        "https://www.data.gouv.fr/fr/datasets/r/fb6c3b2e-da8c-4e69-a719-6a96329e4cb2"
+    )
+    with open(data_dir + "spectacle-download.csv", "wb") as f:
+        for chunk in r.iter_content(1024):
+            f.write(chunk)
+
+    df = pd.read_csv(data_dir + "spectacle-download.csv", dtype=str, sep=";")
+    df = df[df["statut_du_recepisse"] == "Valide"]
+    df["is_entrepreneur_spectacle"] = True
+    df["siren"] = df["siren_personne_physique_siret_personne_morale"].str[:9]
+    df = df[["siren", "is_entrepreneur_spectacle"]]
+    df = df[df["siren"].notna()]
+    df.to_csv(data_dir + "spectacle-new.csv", index=False)
+
+
+def preprocess_uai_data(
+    data_dir,
+) -> None:
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+
+    r = requests.get(
+        "https://www.data.gouv.fr/fr/datasets/r/b22f04bf-64a8-495d-b8bb-d84dbc4c7983"
+    )
+    with open(data_dir + "uai-download.csv", "wb") as f:
+        for chunk in r.iter_content(1024):
+            f.write(chunk)
+
+    df = pd.read_csv(data_dir + "uai-download.csv", dtype=str, sep=";")
+    df = df[["identifiant_de_l_etablissement", "siren_siret", "code_nature"]]
+    df = df.rename(
+        columns={"identifiant_de_l_etablissement": "uai", "siren_siret": "siren"}
+    )
+    df["siren"] = df["siren"].str[:9]
+    res = df.groupby(["siren"])["uai"].apply(list).reset_index(name="liste_uai")
+    res2 = (
+        df.groupby(["siren"])["code_nature"]
+        .apply(list)
+        .reset_index(name="liste_code_nature")
+    )
+    res = res.merge(res2, on="siren", how="left")
+    res = res[["siren", "liste_uai"]]
+    res.to_csv(data_dir + "uai-new.csv", index=False)
+
+
+def compare_versions_file(
+    original_file: str,
+    new_file: str,
+):
+    should_continue = not filecmp.cmp(original_file, new_file)
+    return should_continue
+
+
+def generate_updates_colter(df, current_color):
```

Comment:
The `task_functions.py` file is too large. It might be a good idea to separate the indexation functions and have a file where elastic search documents are generated.